pallets/creditcoin/src/types/loan_terms.rs

```diff
@@ -35,13 +35,22 @@ impl Duration {
 	}
 }
 
+#[derive(Clone, Copy, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+pub enum InterestType {
+	Simple,
+	Compound,
+}
+
 #[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
 #[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
 pub struct InterestRate {
 	pub rate_per_period: RatePerPeriod,
 	pub decimals: Decimals,
 	pub period: Duration,
+	pub interest_type: InterestType,
```

Comment:
Is the scope here only to add the new field and enum without making any use of them ?

Reply:
Yeah, pretty much. We don’t calculate interest on-chain, so this is just to provide more information for users looking to assess credit or figure out the expected loan payment